### ISSUE TITLE
refactor(WebWorker): move web worker to its own module

### ISSUE THREAD

**Issue by jeffbcross** [angular/angular#4064]
(Recapping an IRL convo between me, @jteplitz602 and @alexeagle )

TL;DR: I propose moving all `web_worker` logic into its own module as a sibling of angular2, to be located at `modules/angular2_worker`.

Why?

 * All sources inside of the angular2 module should semantically be considered children of angular2
 * Worker is really a different distribution of angular2, not a child
 * Everything inside the angular2 folder should logically be exportable in a catch-all angular2.js bundle. Technically, the contents of worker happen to be safe to export inside of an angular2.js bundle, but they don't play well with the other children.
 * Worker re-exports much of what is in Angular2. This makes typings generation awkward.

This would also mean distributing it as its own npm and dart package.

CC: @jteplitz602 @yjbanov @alexeagle @mhevery @IgorMinar 

TODO:
- [x] Change public import path to "angular2_worker/worker" & "angular2_worker/ui"
- [x] Move src directory
- [x] Move tests
- [ ] Move manual_typings out of modules & add build step to copy it into correct place
- [x] Move worker test utils out of angular2/test_lib
- [x] Update test.unit.dart, test.unit.dartvm, test.unit.cjs, test.unit.js to run angular2_worker tests
- [ ] Move docs/web_workers
- [x] Create a package.json
- [ ] Copy tsd files
- [x] Change all paths in examples to import from *only* Worker public api
- [ ] Rename WebWorker to Worker
- [x] Generate separate .d.ts files for Worker in typings/angular2_worker/worker.d.ts & typings/angular2_worker/ui.d.ts
- [x] Update typings tests to test worker typings
- [x] Generate separate bundles in angular2_worker/worker.js & angular2_worker/ui.js
- [ ] Publish as separate npm package (scripts/publish/npm_publish.sh)
- [x] Publish as separate dart package (scripts/publish/pub_publish.sh)
- [ ] Publish to code.angularjs.org 
- [ ] Set up 3rd party package (talk to @yjbanov about this)

**Comment by jeffbcross** [comment]
Just discussed IRL with @yjbanov and @jteplitz602 and we agreed this made sense. @jteplitz602 and I are going to start on this later this afternoon, hopefully getting the move completed by Friday.

**Comment by jteplitz602** [comment]
@jeffbcross Do you think the worker npm module should depend on angular2 or redistribute its sources? 
Semantically it feels as if angular2_worker wraps angular2 so it makes sense to think of angular2 as a child of angular2_worker. That's currently the way that I've set up the dart package.

**Comment by jteplitz602** [comment]
@jeffbcross I'm reassigning this to you. I'm nearly done in this branch https://github.com/jteplitz602/angular/tree/move-web-workers. The checklist above is up to date, so you should only need to do the unchecked tasks.

**Comment by jeffbcross** [comment]
@pkozlowski-opensource I closed this issue because I think it's no longer valid, other than work you're doing on bundles. Is that correct?
